### PR TITLE
Add support for multiple admissions

### PR DESCRIPTION
**Pull request by norbye**
Resolves ABA-18

## Summary of changes
* Display all available admissions on the landing page
* Pass admissionId as URL param when navigating between pages
* Update API to receive admission ID for relevant endpoints (applications, my application, group)
* Add model-connection between admission and group
* Return groups through their connected admission

Tests are not passing, but with the new URL structure I don't see the previous system with `reverse()` working very well - so I'll postpone that fix until the changes in the API endpoints are final.

## Updated API-endpoints

These are mostly the same endpoints as previously, just restructured to match the relation between admissions and groups/applications.

<table>
<tr>
	<th>Action
	<th>Path
<tr>
	<td>GET
	<td>/admission
<tr>
	<td>GET
	<td>/admission/{id}
<tr>
	<td>GET
	<td>/admission/{id}/application/
<tr>
	<td>POST
	<td>/admission/{id}/application/
<tr>
	<td>GET
	<td>/admission/{id}/application/mine/
<tr>
	<td>DELETE
	<td>/admission/{id}/application/mine/
<tr>
	<td>DELETE
	<td>/admission/{id}/application/delete_group_application/
<tr>
	<td>PATCH
	<td>/group/{id}
</table>

## As it's only visual change, it displays all admissions on the frontpage
image

### CODE REVIEW

**Comment by AdrianAndersen** [comment]
Would be nice if you could include the linear issue ID (in this case ABA-18) in the description for the PR. That way, it automagically links to the Linear issue, and marks it as done when this is merged.

File: committee_admissions/admissions/tests/test_api.py [hunk 1]
```diff
@@ -71,7 +85,7 @@ def test_pleb_cannot_edit_group(self):
         self.client.force_authenticate(user=self.pleb)
 
         res = self.client.patch(
-            reverse("group-detail", kwargs={"pk": self.arrkom.pk}),
+            self.urls["group-detail"].format(self.arrkom.pk),
```
Comment (LudvigHz) [inline committee_admissions/admissions/tests/test_api.py:88]: You really _should_ use the `reverse` function here and not create the url yourself, but it's not the end of the world.

File: frontend/src/routes/ApplicationPortal.jsx [hunk 2]
```diff
@@ -85,38 +85,53 @@ const ApplicationPortal = () => {
     return <div>Error: {error.message}</div>;
   } else if (isFetching) {
     return <LoadingBall />;
+  } else if (admission?.detail === "Not found.") {
+    return <NotFoundPage />;
```
Comment (LudvigHz) [inline frontend/src/routes/ApplicationPortal.jsx:89]: hmm, I'm guessing you don't have access to the http status code here. That would be alot better than this if possible.

**Review by LudvigHz — APPROVED** [review]
LGTM :shipit: 

Not much to complain about here, nice work :1st_place_medal:

File: frontend/src/index.jsx [hunk 3]
```diff
@@ -57,9 +57,7 @@ const App = ({ children }) => {
 const AppRoutes = () =>
   useRoutes([
     { path: "/", element: <LandingPage /> },
-    { path: "/velg-komiteer", element: <ApplicationPortal /> },
-    { path: "/min-soknad", element: <ApplicationPortal /> },
-    { path: "/admin", element: <ApplicationPortal /> },
+    { path: ":admissionId/*", element: <ApplicationPortal /> },
```
[the remark this answers is not in the document]
Comment (norbye) [reply frontend/src/index.jsx:60]: Definitely agree, will take a look at adding it later

File: committee_admissions/admissions/tests/test_api.py (hunk 1 above)
Comment (norbye) [reply committee_admissions/admissions/tests/test_api.py:88]: Had some problems getting that to work when I used [this approach](https://github.com/webkom/committee-admissions/blob/multiple-admissions/committee_admissions/urls.py#L31) to define the application URL, but I'll take a crack at getting it to work on todays workshop.   
And I'll fix it for the group-detail as that route should still work with the old usage of the reverse method.

**Review by LudvigHz — APPROVED** [review]
LGTM :100:

**Review by eikhr — APPROVED** [review]
I've only looked at the frontend, but it looks very good!

**Pull request merged**